ast/src/parsed/asm.rs

```diff
@@ -292,9 +292,29 @@ pub enum InstructionBody<T> {
     CallableRef(CallableRef),
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum OutputRegister {
```

Comment:
Why "OutputRegister"? Isn't it an assignment register i.e. both an output and an input register?
What about "RegisterForAssignment"?

Reply:
Renamed it to `AssignmentRegister`